eventdemo/source/c/src/kaa_demo.c

```diff
@@ -19,212 +19,146 @@
 #include <string.h>
 #include <time.h>
 
-#include <kaa/kaa_error.h>
-#include <kaa/kaa_context.h>
-#include <kaa/platform/kaa_client.h>
-#include <kaa/utilities/kaa_log.h>
-#include <kaa/utilities/kaa_mem.h>
+#include <kaa_error.h>
+#include <kaa_context.h>
+#include <platform/kaa_client.h>
+#include <utilities/kaa_log.h>
+#include <utilities/kaa_mem.h>
 #include <kaa_user.h>
-#include <kaa/gen/kaa_thermostat_event_class_family.h>
+#include <gen/kaa_chat.h>
 
+#include <pthread.h>
 
 
-#define KAA_USER_ID            "user@email.com"
+#define KAA_USER_ID            "userid"
 #define KAA_USER_ACCESS_TOKEN  "token"
 
-#define THERMO_REQUEST_FQN          "org.kaaproject.kaa.schema.sample.event.thermo.ThermostatInfoRequest"
-#define CHANGE_DEGREE_REQUEST_FQN   "org.kaaproject.kaa.schema.sample.event.thermo.ChangeDegreeRequest"
+#define CHAT_EVENT_FQN     "org.kaaproject.kaa.examples.event.ChatEvent"
+#define CHAT_MESSAGE_FQN   "org.kaaproject.kaa.examples.event.Message"
 
 
+char current_room[100];
 
-static kaa_client_t *kaa_client = NULL;
-static bool is_shutdown = false;
-
-
-
-#define KAA_DEMO_UNUSED(x) (void)(x);
+struct room {
```

Comment:
Is there any reason to introduce `room` type and then add the alias `Room` for it?
`_t` is reserved.